server/src/main/java/io/seata/server/AbstractTCInboundHandler.java

```diff
@@ -91,12 +81,42 @@ protected abstract void doGlobalCommit(GlobalCommitRequest request, GlobalCommit
     @Override
     public GlobalRollbackResponse handle(GlobalRollbackRequest request, final RpcContext rpcContext) {
         GlobalRollbackResponse response = new GlobalRollbackResponse();
-        exceptionHandleTemplate(new Callback<GlobalRollbackRequest, GlobalRollbackResponse>() {
+        exceptionHandleTemplate(new AbstractCallback<GlobalRollbackRequest, GlobalRollbackResponse>() {
             @Override
             public void execute(GlobalRollbackRequest request, GlobalRollbackResponse response)
-                throws TransactionException {
+                    throws TransactionException {
                 doGlobalRollback(request, response, rpcContext);
             }
+
+            @Override
+            public void onTransactionException(GlobalRollbackRequest request, GlobalRollbackResponse response, TransactionException tex) {
+                super.onTransactionException(request, response, tex);
+                try {
+                    GlobalSession globalSession = SessionHolder.findGlobalSession(XID.getTransactionId(request.getXid()));
+                    if (globalSession != null) {
+                        response.setGlobalStatus(globalSession.getStatus());
+                    } else {
+                        response.setGlobalStatus(GlobalStatus.UnKnown);
+                    }
+                } catch (TransactionException e) {
+                    response.setGlobalStatus(GlobalStatus.UnKnown);
+                }
+            }
+
+            @Override
+            public void onError(GlobalRollbackRequest request, GlobalRollbackResponse response, Exception rex) {
+                super.onError(request, response, rex);
+                try {
+                    GlobalSession globalSession = SessionHolder.findGlobalSession(XID.getTransactionId(request.getXid()));
+                    if (globalSession != null) {
+                        response.setGlobalStatus(globalSession.getStatus());
+                    } else {
+                        response.setGlobalStatus(GlobalStatus.UnKnown);
+                    }
+                } catch (TransactionException e) {
+                    response.setGlobalStatus(GlobalStatus.UnKnown);
```

Comment:
you can remove findGlobalSession's throw TransactionException, it never throw TransactionException.
![image](https://user-images.githubusercontent.com/8758457/57755622-ed482600-7723-11e9-90f0-85ec51008ad3.png)

Reply:
ok。I will fix it